astropy/units/utils.py

```diff
@@ -275,10 +294,12 @@ def resolve_fractions(a, b):
     """
     # We short-circuit on the most common cases of int and float, since
     # isinstance(a, Fraction) is very slow for any non-Fraction instances.
-    a_is_fraction = (a.__class__ is not int and a.__class__ is not float and
-                     isinstance(a, Fraction))
-    b_is_fraction = (b.__class__ is not int and b.__class__ is not float and
-                     isinstance(b, Fraction))
+    a_is_fraction = (
+        a.__class__ is not int and a.__class__ is not float and isinstance(a, Fraction)
```

Comment:
```suggestion
        type(a) is not int and type(a) is not float and isinstance(a, Fraction)
```

Reply:
Down to 1 line?

Reply:
I vaguely recall really trying to optimize this and `a.__class__` is slightly faster than `type(a)` -- this definitely is in the hot path. Anyway, in the end it seemed best to just leave it; this is `black` style (and I will admit getting slightly used to it, though I still find it wastes too much vertical space).